pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -0,0 +1,368 @@
+// ClusterVersionOperatorIsDown is firing
+var gatheringRules = []gatheringRule{
+	{
+		Conditions: []conditionWithParams{
+			{
+				Type: alertIsFiring,
+				Params: map[string]interface{}{
+					"name": "SamplesImagestreamImportFailing",
+				},
+			},
+		},
+		GatheringFunctions: map[gatheringFunctionName]GatheringFunctionParams{
+			gatherLogsOfNamespace: map[string]interface{}{
+				"namespace":      "openshift-cluster-samples-operator",
+				"tail_lines":     100,
+				"label_selector": "",
+			},
+			gatherImageStreamsOfNamespace: map[string]interface{}{
+				"namespace": "openshift-cluster-samples-operator",
+			},
+		},
+	},
+}
+
+const canConditionalGathererFail = false
+
+// Gatherer implements the conditional gatherer
+type Gatherer struct {
+	gatherProtoKubeConfig   *rest.Config
+	metricsGatherKubeConfig *rest.Config
+	imageKubeConfig         *rest.Config
+	firingAlerts            map[string]bool // golang doesn't have sets :(
+}
+
+// New creates a new instance of conditional gatherer with the appropriate configs
+func New(gatherProtoKubeConfig, metricsGatherKubeConfig *rest.Config) *Gatherer {
+	imageKubeConfig := rest.CopyConfig(gatherProtoKubeConfig)
+	imageKubeConfig.QPS = common.ImageConfigQPS
+	imageKubeConfig.Burst = common.ImageConfigBurst
+
+	return &Gatherer{
+		gatherProtoKubeConfig:   gatherProtoKubeConfig,
+		metricsGatherKubeConfig: metricsGatherKubeConfig,
+		imageKubeConfig:         imageKubeConfig,
+	}
+}
+
+// GetName returns the name of the gatherer
+func (g *Gatherer) GetName() string {
+	return "conditional"
+}
+
+// GetGatheringFunctions returns gathering functions that should be run considering the conditions
+// + the gathering function producing metadata for the conditional gatherer
+func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gatherers.GatheringClosure, error) {
+	err := g.updateAlertsCache(ctx)
+	if err != nil {
+		klog.Errorf("conditional gatherer can't update alerts cache")
+		return nil, nil
+	}
+
+	gatheringFunctions := make(map[string]gatherers.GatheringClosure)
+
+	gatheringFunctions["conditional_gatherer_rules"] = gatherers.GatheringClosure{
+		Run:     g.GatherConditionalGathererRules,
+		CanFail: canConditionalGathererFail,
+	}
+
+	for _, conditionalGathering := range gatheringRules {
+		allConditionsAreSatisfied, err := g.areAllConditionsSatisfied(conditionalGathering.Conditions)
+		if err != nil {
+			return nil, err
+		}
+
+		if allConditionsAreSatisfied {
+			functions, errs := g.createGatheringClosures(conditionalGathering.GatheringFunctions)
+			if len(errs) > 0 {
+				return nil, err
+			}
+
+			for funcName, function := range functions {
+				gatheringFunctions[funcName] = function
+			}
+		}
+	}
+
+	return gatheringFunctions, nil
+}
+
+// GatherConditionalGathererRules stores the gathering rules in insights-operator/conditional-gatherer-rules.json
+func (g *Gatherer) GatherConditionalGathererRules(context.Context) ([]record.Record, []error) {
```

Comment:
This will go to our `docs/gathered-data.md` as I just noticed :) Not sure we want it there....but probably not a big deal. WDYT @jholecek-rh pls? :)

Reply:
We want it there, imho.

I'm just worried about the terminology. Insights rule developers are part of the primary audience for `docs/gathered-data.md`. They will have to pause to realize that it is not talking about Insights rules (like I did right now). I just cannot think of a better name...

Reply:
Good points @jholecek-rh. Thanks!